### PR TITLE
Remove `make_future` round 2

### PR DESCRIPTION
**Pull request by msimberg**
On top of #537. Related to #468.

Changes `scheduleAllReduceInPlace` and `t_factor`. Quite a few open questions still in the code.

### CODE REVIEW

**Comment by msimberg** [comment]
bors try

**Comment by msimberg** [comment]
bors try

File: include/dlaf/eigensolver/bt_reduction_to_band/impl.h [hunk 1]
```diff
@@ -346,7 +346,7 @@ void BackTransformationReductionToBand<backend, device, T>::call(
     }
 
     for (const auto& kj_panel : panelW2.iteratorLocal())
-      dlaf::comm::scheduleAllReduceInPlace(mpi_col_task_chain(), MPI_SUM, panelW2(kj_panel));
+      dlaf::comm::scheduleAllReduceInPlace<T>(mpi_col_task_chain(), MPI_SUM, panelW2(kj_panel));
```
Comment (msimberg) [inline include/dlaf/eigensolver/bt_reduction_to_band/impl.h:349]: Change to use `SenderElementType` inside `scheduleAllReduceInPlace`.

Comment (msimberg) [reply include/dlaf/eigensolver/bt_reduction_to_band/impl.h]: Done.

File: include/dlaf/communication/kernels/all_reduce.h [hunk 2]
```diff
@@ -134,7 +133,8 @@ pika::future<matrix::Tile<T, Device::CPU>> scheduleAllReduceInPlace(
                             std::bind(copyBack_o, std::placeholders::_1, std::cref(tile))) |
                   ex::then([&tile]() { return std::move(tile); });
          }) |
-         ex::make_future();
+         // TODO: Leave this up to the caller?
+         ex::ensure_started();
```
[the remark this answers is not in the document]
Comment (msimberg) [reply include/dlaf/communication/kernels/all_reduce.h]: I did this change. However, I'd be interested to hear if others like it as well.

**Comment by msimberg** [comment]
bors try

**Comment by msimberg** [comment]
This is likely easier to review once #537 is in.

**Comment by msimberg** [comment]
bors try

**Comment by rasolca** [comment]
Can you please merge/rebase to resolve conflicts and simplify the review operations?

**Comment by rasolca** [comment]
bors cancel

**Comment by rasolca** [comment]
bors ping

**Comment by rasolca** [comment]
bors try

**Comment by msimberg** [comment]
bors ping

**Pull request merged**